modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -224,6 +224,9 @@ void configureGeneratorProperties() {
         generateApiTests = GlobalSettings.getProperty(CodegenConstants.API_TESTS) != null ? Boolean.valueOf(GlobalSettings.getProperty(CodegenConstants.API_TESTS)) : getGeneratorPropertyDefaultSwitch(CodegenConstants.API_TESTS, true);
         generateApiDocumentation = GlobalSettings.getProperty(CodegenConstants.API_DOCS) != null ? Boolean.valueOf(GlobalSettings.getProperty(CodegenConstants.API_DOCS)) : getGeneratorPropertyDefaultSwitch(CodegenConstants.API_DOCS, true);
 
+        // allows disabling the default generation of OpenAPI Generator metadata files.
+        generateMetadata = config.additionalProperties().containsKey(CodegenConstants.GENERATE_METADATA_FILES) ? Boolean.valueOf(config.additionalProperties().get(CodegenConstants.GENERATE_METADATA_FILES).toString()) : getGeneratorPropertyDefaultSwitch(CodegenConstants.GENERATE_METADATA_FILES, true);
```

Comment:
It seems like this `generateMetadata` flag was only ever being used during testing. Not sure why - it's a useful flag to expose to consumers.